source/common/common/url_template_matcher.cc

```diff
@@ -0,0 +1,31 @@
+#include "common/common/url_template_matcher.h"
+
+#include "common/common/empty_string.h"
+#include "common/common/utility.h"
+
+namespace Envoy {
+namespace Matchers {
+
+namespace {
+
+// Http method for all url_template matches.
+// This could be anything.
+static constexpr absl::string_view HTTP_METHOD{"GET"};
+
+} // namespace
+
+UrlTemplateMatcher::UrlTemplateMatcher(const absl::string_view url_template) {
+  google::grpc::transcoding::PathMatcherBuilder<const UrlTemplateMatcher*> pmb;
```

Comment:
I think we need to do a full review ala https://github.com/envoyproxy/envoy/blob/main/DEPENDENCY_POLICY.md#new-external-dependencies before we take a dependency on this extension in the core. Can you fill out this checklist @qiwzhang?

Reply:
https://github.com/envoyproxy/envoy/issues/15304

Reply:
Thanks; I assume there is also some path matcher specific fuzzing to add to that burndown from other conversations, or maybe I'm misremembering.

Reply:
Yes,  I will like to do that in separate PR